node_agent/main.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2017 Istio Authors
```

Comment:
The one I tried to follow is similar to what is used for Kubernetes and pilot, where binaries are placed into a top-level `cmd/` directory. See [here](https://github.com/kubernetes/kubernetes/tree/master/cmd). The main files/methods in the `cmd/` dir mainly deal with CLI parsing while the actually logic is implemented in related packages under `pkg/`.

The reasons that I didn't choose the cobra one are
1. Less top-level directories (otherwise we'll have one top-level directory for each executable we have, which makes this repo hard to navigate IMO).
2. You can avoid having the `na` package insider `node_agent` dir, which is non-ideal.

I agree that the repo needs some re-structure regardless which way to choose. I'll file an issue and we can move the discussion over there.